### DISCUSSION TITLE
What is the right usage of WarmUp and how does it work?

### DISCUSSION THREAD

**Discussion by lukasferda** [VIP-Logic/NBomber#590] [Q&A]
Hi,
Let me point out one blind spot in the documentation:
In the following link, https://nbomber.com/docs/using-nbomber/basic-api/scenario#scenario-warmup, it only explains "how to use it." However, I feel that it lacks a bit more context, such as how it works or what the right use case for it is. 
Regards Lukas F.

**Comment by AntyaDev** [comment]
Hi @lukasferda 

Thanks for the good point and recommendation. We will try to improve it and add more context.

**Comment by StMarian** [comment]
bump

**Comment by AntyaDev** [comment]
Warm-up is needed for several reasons:
- .NET JIT will compile your Scenario code to run the code efficiently. Usually, if you skip warmup, your first scenario iterations/requests will show a bigger MAX latency.
- A warm-up is also necessary to make sure that the state of server resources, such as memory or cache, is prepared for the load testing.

It depends on the use case but you can set at least 3-5 sec for warm-up.
